pkg/restore/restore.go

```diff
@@ -1267,17 +1268,130 @@ func (ctx *restoreContext) restoreItem(obj *unstructured.Unstructured, groupReso
 				_, err = resourceClient.Patch(name, patchBytes)
 				if err != nil {
 					warnings.Add(namespace, err)
+					// check if there is existingResourcePolicy and if it is set to update policy
+					if len(ctx.restore.Spec.ExistingResourcePolicy) > 0 && ctx.restore.Spec.ExistingResourcePolicy == velerov1api.PolicyTypeUpdate {
+						//try patching just the backup/restore labels
+						patchBytes, err := generatePatch(fromCluster, fromClusterWithLabels)
```

Comment:
This will generate an empty patch, since fromCluster has the new label, as does fromClusterWithLabels. You need to call `removeRestoreLabels(fromCluster)` before this. In fact, you probably need to call it before `generatePatch` on line 1254/1255 above so that we also get the updated labels when we succeed in merging the service accounts.

Reply:
Done, fixed for `serviceaccounts`